### DISCUSSION TITLE
Why does the background not change while mouseover in Templated Controls?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#10624] [Q&A] [answered]
What I want to do is change the background to the value of HoverBackground when the mouse is over.

Here is my code:
`<Styles xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:controls="using:test.TemplatedControls">
	<Design.PreviewWith>
		<controls:NormalButton />
	</Design.PreviewWith>
	<Style Selector="controls|NormalButton">
		
		<Setter Property="Background" Value="Blue"></Setter>
		<Setter Property="Template">		
			<ControlTemplate>
				<Border Name="B" Background="{TemplateBinding Background}">
					<ContentPresenter></ContentPresenter>
				</Border>
			</ControlTemplate>
		</Setter>
	</Style>
	<Style Selector="Button:pointerover /template/ ContentPresenter#B">
		<Setter Property="Background" Value="{TemplateBinding HoverBackground}" />
	</Style>
</Styles>
`

And here is code-behind:
`using Avalonia;
using Avalonia.Controls;
using Avalonia.Controls.Primitives;
using Avalonia.Media;
namespace test.TemplatedControls
{
    public class NormalButton : Button
    {
        public static readonly StyledProperty<Brush> HoverBackgroundProperty =
    AvaloniaProperty.Register<Border, Brush>(nameof(HoverBackground));
        public Brush HoverBackground
        {
            get { return GetValue(HoverBackgroundProperty); }
            set { SetValue(HoverBackgroundProperty, value); }
        }
    }
}`

**Comment by rabbitism** [comment, marked as answer]
`(nameof(Background)`
should be the new property name

**Reply by kyuranger** [reply]
Thank you. I fixed this. However, the background still does not change when the mouse is over.

**Reply by rabbitism** [reply]
and in selector `Selector="Button:pointerover`, it should be `Selector="controls|NormalButton:pointerover`

**Reply by kyuranger** [reply]
I changed like this:
<Style Selector="controls|NormalButton:pointerover /template/ ContentPresenter#B">
		<Setter Property="Background" Value="Green" />
	</Style>
It doesn't work any.
After I change like this:
<Style Selector="controls|NormalButton:pointerover">
		<Setter Property="Background" Value="Green" />
</Style>

It works now. How can I change the value target to a control in ControlTemplate(for example border B in my project), just like TargetName property in WPF.

**Reply by kyuranger** [reply]
> and in selector `Selector="Button:pointerover`, it should be `Selector="controls|NormalButton:pointerover`

I got it. It should be:
	<Style Selector="controls|NormalButton:pointerover /template/ Border#B">
		<Setter Property="Background" Value="Green" />
	</Style>

Thank you.谢谢
